### DISCUSSION TITLE
evtx2es how to add event.dataset and event.module fields

### DISCUSSION THREAD

**Discussion by zaphode** [Security-Onion-Solutions/securityonion#3406] [Unsupported Versions] [answered]
SOLVED: 
[Installing SO Version of evtx2es](https://docs.google.com/document/d/12TGO_fHNQb1gvov0UUEu90_TfH43480SnZEfC_2csQo/edit?usp=sharing)

I downloaded the updated version of evtx2es and it is working perfect so far.
One thing I am trying to figure out is how to get the import to add the event.dataset and 
event.module fields so the imported logs can be included in the so Kibana Home console.

The Windows logs show up like this when ingested in the live system. I am guessing these are getting parsed and the fields added after the evtx2es script sends them to Elasticsearch. Is there somewhere that I need to tell elastic search to create those fields?  

- **event.module: "windows_eventlog"** 
- **event.dataset: "security", "alert", "system" ,"application"**

**so-import-pcap** does have these fields but I am guessing that it is because zeek and suricata are analyzing them and adding those fields. 

I am hoping this is not a dumb question, I may be misunderstanding how this is getting processed. :)~

**Comment by defensivedepth** [comment, marked as answer]
All the normal caveats apply here - this functionality is experimental and still under development and will probably change in the near future etc etc.

With that being said - try adding the following parameter when invoking evtx2es: `--pipeline import.wel`

This directs evtx2es to use an Ingest Pipeline that is specifically configured for importing Windows eventlogs.

**Reply by zaphode** [reply]
That did the trick :)
Thanks

Here is what ended up working. 
**`evtx2es --index so-evtx2es --pipeline import.wel <filename.evtx>`**

**Reply by defensivedepth** [reply]
Everything seem to be working as you would expect at this point? Fields renamed and parsed correct, etc?
